### PR TITLE
Define path with __dir__

### PR DESCRIPTION
**Pull request by DirtyF**
Inspired by https://bogdanvlviv.github.io/posts/ruby/with-__dir__-we-can-restore-order-in-the-universe.html

/cc @bogdanvlviv

### CODE REVIEW

File: lib/jekyll/assets.rb [hunk 1]
```diff
@@ -13,7 +13,7 @@
 ]
 
 requires.each do |part|
-  Pathutil.new(File.expand_path("../", __FILE__)).join("assets", part.to_s)
+  Pathutil.new(File.expand_path(__dir__).join("assets", part.to_s)
```
Comment (bogdanvlviv) [inline lib/jekyll/assets.rb]: looks like there is missed `)`

**Comment by bogdanvlviv** [comment]
Could you `squash` the commits

**Comment by DirtyF** [comment]
@bogdanvlviv done.

**Comment by DirtyF** [comment]
@bogdanvlviv yeah, it's up to the maintainer now. Thanks for the review. 
All official Jekyll plugins should soon all use `__dir__` ✊

File: lib/jekyll/assets.rb [hunk 2]
```diff
@@ -13,7 +13,7 @@
 ]
 
 requires.each do |part|
-  Pathutil.new(File.expand_path("../", __FILE__)).join("assets", part.to_s)
+  Pathutil.new(File.expand_path(__dir__)).join("assets", part.to_s)
```
Comment (ixti) [inline lib/jekyll/assets.rb]: There's no need in `expand_path`. Use `__dir__` directly:

``` ruby
Pathutil.new(__dir__).join("assets", part.to_s)
```

Comment (DirtyF) [reply lib/jekyll/assets.rb]: fixed.

File: lib/jekyll/assets.rb [hunk 3]
```diff
@@ -13,8 +13,8 @@
 ]
 
 requires.each do |part|
-  Pathutil.new(File.expand_path("../", __FILE__)).join("assets", part.to_s)
-    .glob("{*,**/*}.rb").map(&method(
-      :require
-    ))
+  Pathutil.new(__dir__).join('assets', part.to_s)
+          .glob('{*,**/*}.rb').map(&method(
```
Comment (envygeeks) [inline lib/jekyll/assets.rb]: Can this tabbing change be corrected.

Comment (DirtyF) [reply lib/jekyll/assets.rb]: Fixed.

**Comment by envygeeks** [comment]
Thanks :heart:

**Pull request merged**